genericapi/handler.go

```diff
@@ -89,6 +89,17 @@ func (h *Handler) ServeCreateAlert(w http.ResponseWriter, r *http.Request) {
 	details := r.FormValue("details")
 	action := r.FormValue("action")
 	dedup := r.FormValue("dedup")
+	metaData := r.FormValue("meta")
+	var md alert.AlertMetaInput
+	var meta alert.AlertMeta
+	if metaData != "" {
+		err = json.Unmarshal([]byte(metaData), &md)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		meta = alert.ToAlertMeta(md)
+	}
```

Comment:
We don't want to encode JSON inside the URL or form values, let's stick to form encoding

```suggestion

	metaData := make(alert.Metadata)
	for _, v := range r.Form["meta"] {
		key, val, ok := strings.Cut(s, "=")
		if !ok {
			continue
		}
		metaData[key] = val
	}
```